### ISSUE TITLE
Feature Request:  AUTO Button for Curves and Levels

### ISSUE THREAD

**Issue by ErnestWeber** [photopea/photopea#3572]
It is a handy tool and I use it often.  When you know about how to change the algorithm it can be a powerful tool and so, a worthwhile addition.  

I am not sure how many people know how useful the various algorithms are and that changing them can yield different results, but its really a must have for editing curves and levels.  It has saved me a few times when it came to color or dark and light adjustments when I had the right algorithm and then used the eyedropper from there.

 If possible, it would be great to see the AUTO feature added to Photopea in a future version in both types of adjustment layers, (Curves and Levels) with the ability to set the algorithm via: Auto Color Correction Options which look like:

[ ] Enhance Monochromatic Contrast
[ ] Enhance Per Channel Contrast
[ ] Find Dark & Light Colors
[ ] Enhance Brightness and Contrast

[ ] Snap Neutral Midtones

(target colors & clipping)

etc.

Thanks for considering it.

**Comment by photopea** [comment]
Hi, I have implemented it. Both Curves and Levels have the "Auto" button.

I came up only with the first two algorithms. I could not come up with analogies to the last two algorithms, so they work a lot differently than in Photoshop now.

**Comment by ErnestWeber** [comment]
Oh, that's great!  I didn't expect it so soon and thanks so much for fulfilling my request!
Photopea is just remarkable and I encourage any improvements or updates.

    On Saturday, September 18, 2021, 04:04:26 PM EDT, Photopea ***@***.***> wrote:  
 
 


Hi, I have implemented it. Both Curves and Levels have the "Auto" button.

I came up only with the first two algorithms. I could not come up with analogies to the last two algorithms, so they work a lot differently than in Photoshop now.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
